### PR TITLE
Pasting a copied file or dir provides full path to resource

### LINKED ISSUE #1285: Cmd+v file path into hyper shell
- **OS version and name**: MacOS 10.12.2
- **Hyper.app version**: 1.0.0.1303

## Request
Mac terminal allows the user to cmd+c a directory or file and cmd+v the path to said directory or file into a terminal shell. I use this feature extensively to quickly move between directories that I'm too lazy to get to via command line. It would be helpful if Hyper.app had this feature.

### PR DESCRIPTION
**Pull request by derrickpelletier**
closes #1285 

**ready for merge, tested on windows, linux, and mac**

~not ready for merge because: 
I've tested on linux, and mac, *but not windows* (posted in slack but didn't get a reply), so if someone can test that, that would be great. There is windows specific parsing in there that I haven't tested at all.~

Also, wasn't 100% sure if this is the appropriate spot for this code, but it seemed like it to me (first contrib to `hyper`). Kept clipboard processing in a util file to not clutter the `Term` component. If suggestions for a better place for this, let me know.

### CODE REVIEW

**Comment by derrickpelletier** [comment]
Alright, managed to get this tested on windows. Is ready for review and merge.

File: lib/components/term.js [hunk 1]
```diff
@@ -110,6 +116,17 @@ export default class Term extends PureComponent {
     this.fitResize();
   }
 
+  onWindowPaste(e) {
+    if (!this.props.isTermActive) return;
+
+    const processed = processClipboard();
```
Comment (albinekb) [inline lib/components/term.js]: I'd like a comment to why this code is here to intercept the event, so it's easy to know why this is needed without following the code all the way

File: lib/utils/paste.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import {clipboard} from 'electron';
+
+const getPath = platform => {
+  switch (platform) {
+    case 'darwin': {
+      const filepath = clipboard.read('public.file-url');
```
Comment (albinekb) [inline lib/utils/paste.js]: does this return an empty string if you have something else than a file URL in the clipboard? 🤔  
that is, will this code only intercept pasted file urls and nothing else? (not messing with pasted commands for example)

Comment (derrickpelletier) [reply lib/utils/paste.js]: Yeah, in my tests it returned an empty string. So an empty string would return out of the `processClipboard` method, where term would evaluate to false (no processing happened) and continue with paste. 

I could only see it messing with some non-file paste if some application writes to those specific formats during a copy?

File: lib/components/term.js (hunk 1 above)
Comment (derrickpelletier) [reply lib/components/term.js]: Yep will do

Comment (albinekb) [reply lib/components/term.js]: Perfect 👍

**Comment by derrickpelletier** [comment]
Excellent!

**Comment by jq-87** [comment]
Any idea when this will make it to stable?

**Comment by chabou** [comment]
Very soon. I'm pretty confident that our `2.0.0-canary.14` was our last 2.0.0 canary :tada:

**Pull request merged**